x-pack/plugins/infra/server/routes/infra/lib/mapper.ts

```diff
@@ -32,16 +34,23 @@ export const mapToApiResponse = (
     };
   }
 
-  const hosts = buckets.map((bucket) => {
-    const metrics = convertMetricBucket(params, bucket);
-    const metadata = convertMetadataBucket(bucket);
+  const hosts = buckets
+    .map((bucket) => {
+      const metrics = convertMetricBucket(params, bucket);
+      const metadata = convertMetadataBucket(bucket);
 
-    return { name: bucket.key as string, metrics, metadata };
-  });
+      const cpuValue = metrics.find((metric) => metric.name === 'cpu')?.value ?? 0;
+      const alerts = alertsCountResponse?.find((item) => item.name === bucket.key);
+
+      return { name: bucket.key as string, metrics, metadata, cpuValue, ...alerts };
+    })
+    .sort((a, b) => {
+      return b.cpuValue - a.cpuValue;
+    });
```

Comment:
Up to you, but I'd do it here:
```suggestion
    }).map(({ cpuValue, ...rest }) => rest);
```